specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,640 @@
+# Make MSIX API in Windows App SDK
+
+# Background
+
+The Make MSIX api allows developers to create app packages for distribution. The api
+provides similar functionality as multiple existing command line tools.
+
+For more details see:
+
+-   [The makeappx command line tool](https://learn.microsoft.com/en-us/windows/win32/appxpkg/make-appx-package--makeappx-exe-)
+    Shipped as part of the Windows Development Kit.
+    Exposes pack, unpack, bundle, and unbundle commands, as well as encryption support
+    via a custom key-file list file, and mapping .
+
+-   [The makemsix command line tool and dll](https://github.com/Microsoft/msix-packaging)
+    Available through the github project.
+    Exposes pack, unpack, unbundle, and bundle* commands (*Bundle command is not implemented).
```

Comment:
msix-packaging's makemsix has a bundle command but it does nothing?

Reply:
Well, I spell it out more clearly later, maybe this is overstating the issue. Only flat bundles are supported. Maybe that's what everyone uses now? The failure when trying to create a non-flat bundle is definitely not well documented by the tool, it's just an E_NOTIMPL failure in code that gets ignored and then the user sees a failure message "The bundle must contain at least one app package targeting a known processor architecture." because it never actually processed any of the payload packages:
https://github.com/microsoft/msix-packaging/blob/master/src/msix/pack/AppxBundleWriter.cpp#L117